### PR TITLE
Paging, sort, filter etc

### PR DESCRIPTION
**Pull request by RoriCremer**
Currently not using react virtualize--but a placeholder until then
lNOoIlA32s

### CODE REVIEW

File: src/components/table/DatasetTable.jsx [hunk 1]
```diff
@@ -37,11 +39,24 @@ const styles = theme => ({
 class DatasetTable extends React.PureComponent {
   static propTypes = {
     classes: PropTypes.object.isRequired,
-    rows: PropTypes.array,
+    datasetCount: PropTypes.number,
+    datasets: PropTypes.object.isRequired,
```
Comment (thathert) [inline src/components/table/DatasetTable.jsx]: Should this be an array?

File: src/components/table/DatasetTable.jsx [hunk 2]
```diff
@@ -75,10 +90,23 @@ class DatasetTable extends React.PureComponent {
             <AddSVG className={classes.plusButton} />
           </NavLink>
         </div>
-        <JadeTable columns={columns} rows={rows} />
+        <JadeTable
+          columns={columns}
+          handleFilter={this.handleFilterDatasets}
+          rows={datasets}
+          summary={summary}
+          totalCount={datasetCount || datasets.length}
```
Comment (thathert) [inline src/components/table/DatasetTable.jsx]: I'd prefer if we just used the count, it should always be included from the API.

File: src/components/table/JadeTable.jsx [hunk 3]
```diff
@@ -4,98 +4,184 @@ import { withStyles } from '@material-ui/core/styles';
...
+  };
 
-    this.setState({ order: newOrder, orderBy: property });
+  handleSearchString = event => {
+    const { handleFilter } = this.props;
+    const { page, orderDirection, orderBy, rowsPerPage } = this.state; // limit
+    const offset = page * rowsPerPage;
+    const searchString = event.target.value;
+    this.setState({ searchString }); // filter
+    handleFilter(rowsPerPage, offset, orderBy, orderDirection, searchString);
```
Comment (thathert) [inline src/components/table/JadeTable.jsx]: This is not specific to just filtering -- let's call it something like `handleEnumeration`?

File: src/components/table/JadeTable.jsx [hunk 4]
```diff
@@ -4,98 +4,184 @@ import { withStyles } from '@material-ui/core/styles';
...
   };
 
   render() {
-    const { classes, columns, rows } = this.props;
-    const { order, orderBy } = this.state;
+    const { classes, columns, rows, summary, totalCount } = this.props;
+    const { orderBy, orderDirection, page, rowsPerPage } = this.state;
+    const emptyRows = rowsPerPage - Math.min(rowsPerPage, 8 - page * rowsPerPage);
```
Comment (thathert) [inline src/components/table/JadeTable.jsx]: magic number

File: src/components/table/JadeTable.jsx [hunk 5]
```diff
@@ -4,98 +4,184 @@ import { withStyles } from '@material-ui/core/styles';
...
+              onChange={this.handleSearchString}
+            />
+          </div>
+        )}
+        <Paper className={classes.root}>
+          <Table className={classes.table}>
+            <JadeTableHead
+              columns={columns}
+              onRequestSort={this.handleRequestSort}
+              order={orderDirection}
```
Comment (thathert) [inline src/components/table/JadeTable.jsx]: Let's make this `orderDirection={orderDirection}`

I'm getting: 
```
The prop `orderDirection` is marked as required in `JadeTableHead`, but its value is `undefined`.
```

File: src/components/table/JadeTable.jsx [hunk 6]
```diff
@@ -4,98 +4,184 @@ import { withStyles } from '@material-ui/core/styles';
...
+                <TableRow key={row.id} className={classes.row}>
+                  {columns.map(col => (
+                    <TableCell key={col.property}>
+                      {col.render ? col.render(row) : row[col.property]}
+                    </TableCell>
+                  ))}
+                </TableRow>
+              ))}
+              {rows.length < rowsPerPage && (
+                <TableRow style={{ height: 50 * emptyRows }}>
```
Comment (thathert) [inline src/components/table/JadeTable.jsx]: magic number

File: src/components/table/StudyTable.jsx [hunk 7]
```diff
@@ -58,22 +73,35 @@ class StudyTable extends React.PureComponent {
...
     return (
       <div className={classes.wrapper}>
         <div className={classes.header}>{studyListName || 'STUDIES'}</div>
-        <JadeTable columns={columns} rows={rows} />
+        <JadeTable
+          columns={columns}
+          rows={studies}
+          handleFilter={this.handleFilterStudies}
+          summary={summary}
+          totalCount={studyCount || studies.length}
```
Comment (thathert) [inline src/components/table/StudyTable.jsx]: We should always have studyCount.

File: src/sagas/repository.js [hunk 8]
```diff
@@ -136,10 +136,17 @@ export function* createDataset() {
...
+  const offset = payload.offset || 0;
+  const limit = payload.limit || 5;
+  const filter = payload.searchString || '';
+  const sort = payload.sort || 'created_date';
+  const direction = payload.direction || 'desc';
   try {
-    const response = yield call(authGet, '/api/repository/v1/datasets');
+    const response = yield call(
+      authGet,
+      `/api/repository/v1/datasets?offset=${offset}&limit=${limit}&sort=${sort}&direction=${direction}&filter=${filter}`,
```
Comment (thathert) [inline src/sagas/repository.js]: Can't we pass an object with params into authGet/axios instead of doing the string interpolation ourselves here?

**Review by thathert — CHANGES_REQUESTED** [review]
Looks good! I have a few requests here. Also can we remove the <div> that is inside of the `<Switch>` component so we stop getting the computedMatch error? Thanks :)

File: src/components/table/JadeTable.jsx [hunk 9]
```diff
@@ -4,98 +4,184 @@ import { withStyles } from '@material-ui/core/styles';
...
+                  <TableCell colSpan={columns.length} />
+                </TableRow>
+              )}
+            </TableBody>
+          </Table>
+          {!summary && (
+            <TablePagination
+              rowsPerPageOptions={[5, 10, 25]}
```
Comment (ahaessly) [inline src/components/table/JadeTable.jsx]: magic numbers?

**Review by ahaessly — APPROVED** [review]
👍

File: src/components/table/JadeTable.jsx (hunk 6 above)
Comment (RoriCremer) [reply src/components/table/JadeTable.jsx]: do I want to pass the styling theme into the react component?

File: src/components/table/JadeTable.jsx (hunk 9 above)
Comment (RoriCremer) [reply src/components/table/JadeTable.jsx]: These feel like standard paging choices to me--but you're right, we could enum them somewhere.

**Comment by RoriCremer** [comment]
The rest of the comments will be addressed in a follow on pr

**Review by thathert — APPROVED** [review]

File: src/components/table/JadeTable.jsx (hunk 6 above)
Comment (thathert) [reply src/components/table/JadeTable.jsx]: I don't think so unless we need it inside. I was just thinking of defining this as `ROW_HEIGHT` or something up above.

**Pull request merged**